pkg/profilecreator/profilecreator.go

```diff
@@ -255,8 +256,37 @@ func (ghwHandler GHWHandler) SortedTopology() (*topology.Info, error) {
 	return topologyInfo, nil
 }
 
+func (ghwHandler GHWHandler) TopologyHTDisabled(info *topology.Info) *topology.Info {
+	disabledHTTopology := &topology.Info{
+		Architecture: info.Architecture,
+	}
+	newNodes := []*topology.Node{}
+	for _, node := range info.Nodes {
+		var newNode *topology.Node
+		cores := []*cpu.ProcessorCore{}
+		for _, processorCore := range node.Cores {
+			newCore := cpu.ProcessorCore{ID: processorCore.ID,
+				Index:      processorCore.Index,
+				NumThreads: 1,
+			}
+			for id, logicalProcessor := range processorCore.LogicalProcessors {
+				if id == 0 {
+					newCore.LogicalProcessors = []int{logicalProcessor}
+					cores = append(cores, &newCore)
```

Comment:
I'm not completely sure of the logic here.
Of course we will need to delete some virtual cores, that's fine, but why we are keeping those whose `id == 0` ?
Do we depend on ghw behaviour, on kernel representation or what? This rationale should be made explicit with a code comment.

Reply:
In cases the user wants to disable hyperthreading, while accounting reserved and isolated CPUs we are keeping only the first hyperthread sibling. Example in case in the system there are cores such as {0, 40}, {1,41}, {2, 42}, {3, 43} .... we change that to {0}, {1}, {2}, {3}.... and then do the accounting of reserved and isolated CPU. Will add a comment to explain the rationale.

Reply:
Also, the topologyinfo received here is already sorted by numa ids and cpu ids on the host system.

Reply:
ack, please make explicit if we depend on ghw representation/behaviour.